go.mod

```diff
@@ -0,0 +1,73 @@
+	github.com/coreos/go-semver v0.2.0 // indirect
+	github.com/fd/go-nat v1.0.0 // indirect
+	github.com/go-check/check v0.0.0-20180628173108-788fd7840127 // indirect
+	github.com/gogo/protobuf v1.2.0
+	github.com/google/uuid v1.1.0 // indirect
+	github.com/hashicorp/golang-lru v0.5.0
+	github.com/ipfs/go-cid v0.9.0
+	github.com/ipfs/go-datastore v3.2.0+incompatible
+	github.com/ipfs/go-detect-race v1.0.1 // indirect
+	github.com/ipfs/go-ipfs-util v1.2.8
+	github.com/ipfs/go-log v1.5.7
+	github.com/ipfs/go-todocounter v1.0.1
+	github.com/jbenet/go-cienv v0.0.0-20150120210510-1bb1476777ec // indirect
+	github.com/jbenet/go-context v0.0.0-20150711004518-d14ea06fba99
+	github.com/jbenet/go-randbuf v0.0.0-20160322125720-674640a50e6a // indirect
+	github.com/jbenet/go-temp-err-catcher v0.0.0-20150120210811-aac704a3f4f2 // indirect
+	github.com/jbenet/goprocess v0.0.0-20160826012719-b497e2f366b8
+	github.com/kr/pretty v0.1.0 // indirect
+	github.com/libp2p/go-addr-util v2.0.7+incompatible // indirect
+	github.com/libp2p/go-buffer-pool v0.1.1 // indirect
+	github.com/libp2p/go-conn-security v0.1.15 // indirect
+	github.com/libp2p/go-conn-security-multistream v0.1.15 // indirect
+	github.com/libp2p/go-flow-metrics v0.2.0 // indirect
+	github.com/libp2p/go-libp2p v6.0.29+incompatible
+	github.com/libp2p/go-libp2p-blankhost v0.3.15 // indirect
+	github.com/libp2p/go-libp2p-crypto v2.0.1+incompatible
+	github.com/libp2p/go-libp2p-host v3.0.15+incompatible
+	github.com/libp2p/go-libp2p-interface-connmgr v0.0.21 // indirect
+	github.com/libp2p/go-libp2p-interface-pnet v3.0.0+incompatible // indirect
+	github.com/libp2p/go-libp2p-kbucket v2.2.12+incompatible
+	github.com/libp2p/go-libp2p-loggables v1.1.24 // indirect
+	github.com/libp2p/go-libp2p-metrics v2.1.7+incompatible // indirect
+	github.com/libp2p/go-libp2p-nat v0.8.8 // indirect
+	github.com/libp2p/go-libp2p-net v3.0.15+incompatible
+	github.com/libp2p/go-libp2p-netutil v0.4.12 // indirect
+	github.com/libp2p/go-libp2p-peer v2.4.0+incompatible
+	github.com/libp2p/go-libp2p-peerstore v2.0.6+incompatible
+	github.com/libp2p/go-libp2p-protocol v1.0.0
+	github.com/libp2p/go-libp2p-record v4.1.7+incompatible
+	github.com/libp2p/go-libp2p-routing v2.7.1+incompatible
+	github.com/libp2p/go-libp2p-secio v2.0.17+incompatible // indirect
+	github.com/libp2p/go-libp2p-swarm v3.0.22+incompatible
+	github.com/libp2p/go-libp2p-transport v3.0.15+incompatible // indirect
+	github.com/libp2p/go-libp2p-transport-upgrader v0.1.16 // indirect
+	github.com/libp2p/go-maddr-filter v1.1.10 // indirect
+	github.com/libp2p/go-mplex v0.2.30 // indirect
+	github.com/libp2p/go-msgio v0.0.6 // indirect
+	github.com/libp2p/go-reuseport-transport v0.2.0 // indirect
+	github.com/libp2p/go-stream-muxer v3.0.1+incompatible // indirect
+	github.com/libp2p/go-tcp-transport v2.0.16+incompatible // indirect
+	github.com/libp2p/go-testutil v1.2.10
+	github.com/minio/sha256-simd v0.0.0-20190117184323-cc1980cb0338 // indirect
```

Comment:
I haven't reviewed all dependencies, just random ones. But this one points to the current tip of master at the time of writing (https://github.com/minio/sha256-simd/commit/cc1980cb03383b1d46f518232672584432d7532d) instead of pointing to the commit that the gx version had snapshotted.

I'm not sure if this is intentional or inadvertent. If it's intentional, I suggest we change the title of this PR to `add go mod + upgrade dependencies` to make it explicit for traceability. Otherwise things might start failing beneath our feet if there are new bugs upstream.

We might want to enumerate which dependencies we're upgrading too.

Reply:
I don't think this matters.